test/inductor/test_cpu_repro.py

```diff
@@ -1605,6 +1608,49 @@ def test_auto_simd(self):
                 isa = codecache.pick_vec_isa()
                 self.assertTrue(isa == vec_avx2)
 
+        with config.patch({"cpp.simdlen": None}):
+            os.environ["ATEN_CPU_CAPABILITY"] = "avx2"
+            isa = codecache.pick_vec_isa()
+            if vec_avx512 in codecache.valid_vec_isa_list():
+                self.assertTrue(isa == vec_avx2)
+            elif vec_avx2 in codecache.valid_vec_isa_list():
+                self.assertTrue(isa == vec_avx2)
+            os.environ.pop("ATEN_CPU_CAPABILITY")
+
+        with config.patch({"cpp.simdlen": None}):
+            os.environ["ATEN_CPU_CAPABILITY"] = "avx512"
+            isa = codecache.pick_vec_isa()
+            if vec_avx512 in codecache.valid_vec_isa_list():
+                self.assertTrue(isa == vec_avx512)
+            else:
+                self.assertTrue(isa == vec_avx2)
+            os.environ.pop("ATEN_CPU_CAPABILITY")
+
+        with config.patch({"cpp.simdlen": None}):
+            os.environ["ATEN_CPU_CAPABILITY"] = "default"
+            isa = codecache.pick_vec_isa()
+            self.assertFalse(isa)
+            os.environ.pop("ATEN_CPU_CAPABILITY")
+
+        with config.patch({"cpp.simdlen": None}):
+            os.environ["ATEN_CPU_CAPABILITY"] = "neon"
+            isa = codecache.pick_vec_isa()
+            if vec_avx512 in codecache.valid_vec_isa_list():
+                self.assertTrue(isa == vec_avx512)
+            else:
+                self.assertTrue(isa == vec_avx2)
+            os.environ.pop("ATEN_CPU_CAPABILITY")
+
+        with config.patch({"cpp.simdlen": None}):
+            os.environ["ATEN_CPU_CAPABILITY"] = "zvector"
+            isa = codecache.pick_vec_isa()
+            if vec_avx512 in codecache.valid_vec_isa_list():
+                self.assertTrue(isa == vec_avx512)
+            else:
+                self.assertTrue(isa == vec_avx2)
+            os.environ.pop("ATEN_CPU_CAPABILITY")
+        os.environ["ATEN_CPU_CAPABILITY"] = pre_var
```

Comment:
This should be in a `try-finally` block, otherwise it will leave the old value in place if the test fails.

Reply:
Thanks for your comments. I will use `try-finally`.